### PR TITLE
[GNA] fixed klocwork issues for 2021.4.1

### PR DESCRIPTION
**Pull request by dmitriikhurtin**
### Details:

- fixed klocwork issues for 2021.4.1

### Tickets:
 - CVS-63156

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 1]
```diff
@@ -1752,9 +1752,13 @@ void GNAGraphCompiler::ConvolutionFilterPrimitive(InferenceEngine::CNNLayerPtr l
     void* ptr_biases = nullptr;
 
     IE_ASSERT(!layer->outData.empty());
-    IE_ASSERT(!layer->insData.empty());
     auto outputs = *layer->outData.begin();
-    auto inputs = layer->insData.begin()->lock();
+    auto inputs_iter = layer->insData.begin();
+    if (inputs_iter == layer->insData.end()) {
+        THROW_GNA_EXCEPTION << "The layer '" << layer->name << "' has not inputs";
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1758]: "has no inputs"

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 2]
```diff
@@ -1370,13 +1370,13 @@ void InsertSplitAligningFilterPass::run() {
 #endif
                     auto filterLayer =
                             std::make_shared<ConvolutionLayer>(LayerParams({filterName, "ConvolutionFilter", Precision::FP32}));
+                    if (!filterLayer) {
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:1373]: why is THROW_GNA_EXCEPTION here better than IE_ASSERT?

File: inference-engine/thirdparty/clDNN/src/gpu/custom_task_arena.cpp [hunk 3]
```diff
@@ -115,7 +115,7 @@ void binding_observer::on_scheduler_exit(bool) {
 }
 
 binding_oberver_ptr construct_binding_observer(tbb::task_arena& ta, int num_slots, const constraints& c) {
-    binding_oberver_ptr observer{};
+    binding_oberver_ptr observer{nullptr, binding_observer_deleter()};
```
Comment (elilobanova) [inline inference-engine/thirdparty/clDNN/src/gpu/custom_task_arena.cpp:118]: should we really modify this code which seems not to related to GNA?

File: inference-engine/src/gna_plugin/backend/gna_limitations.hpp [hunk 4]
```diff
@@ -31,7 +31,8 @@ inline bool IsTransposeSupported(const std::vector<size_t>& shape) {
     auto shape_no_1 = shape;
     shape_no_1.erase(std::remove(shape_no_1.begin(), shape_no_1.end(), 1), shape_no_1.end());
     if (shape_no_1.size() != 2) return false;
-    size_t min, max;
+    size_t min = 0;
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/backend/gna_limitations.hpp:34]: is there a possible case when min and max will be still uninitialized after std::tie(min, max) = ... construction?

Comment (dmitriikhurtin) [reply inference-engine/src/gna_plugin/backend/gna_limitations.hpp:34]: The static analyzer gives an error when using uninitialized min and max in std::tie(min, max) = ...

**Pull request closed without merging**